MapboxCoreNavigation/MBRouteController.m

```diff
@@ -30,3 +34,16 @@ - (NSString * _Nonnull)md5 {
     return  output;
 }
 @end
+
+@implementation MBNavigator (additions)
+
+- (NSUInteger)setupRouter:(NSString *)tilesPath translationsPath:(NSString *)translationsPath {
+    
+    @try {
+        return [self configureRouterForTilesPath:tilesPath translationsPath:translationsPath];
+    } @catch (NSException *exception) {
+        return -1;
```

Comment:
It feels like we’re trying to run around unexpected exceptions coming out of MapboxNavigationNative. If `-configureRouterForTilesPath:translationsPath:` has expected failure modes, it should be the responsibility of MapboxNavigationNative to catch an exception and turn it into a C-style return code and NSError out parameter.

By default, [throwing an exception in Objective-C causes a leak](http://clang.llvm.org/docs/AutomaticReferenceCounting.html#exceptions), even under ARC, because the expectation is that an exception is a catastrophic error that leads to an immediate crash anyways. It doesn’t leak in Objective-C++, but whether the `@throw` happens in Objective-C or Objective-C++ is none of this SDK’s business.

Reply:
Returning NSErrors from Obj-C has a really poor bridgeability to Swift’s "throws". Noticed that this try-catch block doesn't suppress the underlying exception. It's just a fallback for the kind of resource the offline router is looking for. (1: Uncompress tiles, 2: Compressed tiles, 3: Different structure of compressed tiles). It's not a fatal error but one that gets caught if you have "Catch all Exceptions" enabled.

Reply:
> Returning NSErrors from Obj-C has a really poor bridgeability to Swift’s "throws".

Unless I’m mistaken, the most idiomatic method signature would include an `NSError **` out parameter, which would bridge to Swift as a `throws`, but it wouldn’t return an error.

Reply:
> the most idiomatic method signature would include an NSError ** out parameter

this is correct but there are some limitations, a boolean value is the only supported primitive return type:
```objc
- (void)voidNoThrow:(int)arg error:(NSError **)error;      // Bridges to NSErrorPointer
- (NSUInteger)intNoThrow:(int)arg error:(NSError **)error; // Bridges to NSErrorPointer
- (BOOL)boolThrow:(int)arg error:(NSError **)error;        // Throws idiomatically
```

However, there seems to be a consensus about using `-fno-exceptions` eventually, partially due to bloated binary size but the inevitable leak in Obj-C is also a good point. Either way, this try-catch block is already removed.